examples/02-javascript-basics/array_manipulation_library.js

```diff
@@ -11,12 +11,62 @@ Join - join the array elements into a string given a parameter
 */
 
 const $ = {
-	filter: function () {},
-	map: function () {},
-	reduce: function () {},
-	head: function () {},
-	tail: function () {},
-	join: function () {}
+	// Filter function
+	filter: function (values, test) {
+		var result = [];
+		for(var i = 0; i < values.length; i++) {
+			if(test(values[i])) {
+				result.push(values[i]);
+			}	
+		};
+
+		return result;
+	},
+	map: function (values, test) {
+		var result = [];
+		for(var i = 0; i < values.length; i++) {
+			result.push(test(values[i]));
+		}
+
+		return result;
+	},
+	reduce: function (values, test, initial_value) {
+		var result = [];
+		var total = 0;
+
+		for(var i = 0; i < values.length; i++) {
+			total = test(values[i], total);
+		}
+		
+		// Checks if there is initial value
+		if(initial_value || initial_value == 0) {
+			total = total + initial_value
+		} else {
+			if(values.length == 0) {
```

Comment:
This seems to be an `else if {}`. Prefer the syntax `else if {}`  when applicable.

Reply:
Noted and applied :)